From eefbb22a1e37d7b3d9cd8c50e6e6285f43865ea2 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:15 +0100
Subject: [PATCH 20/54] media: intel/ipu4: Simplify DMA buffer deallocation
 with dma_free_coherent()

Replace dma_free_attrs() calls with the simpler dma_free_coherent() API
where no special DMA attributes are needed. This improves code clarity and
reduces API surface complexity.
---
 drivers/media/pci/intel/ipu-cpd.c  |  8 +++-----
 drivers/media/pci/intel/ipu-isys.c | 15 ++++++---------
 drivers/media/pci/intel/ipu-psys.c |  8 ++++----
 3 files changed, 13 insertions(+), 18 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-cpd.c b/drivers/media/pci/intel/ipu-cpd.c
index 3ee942778fec..4c0e048290c2 100644
--- a/drivers/media/pci/intel/ipu-cpd.c
+++ b/drivers/media/pci/intel/ipu-cpd.c
@@ -232,10 +232,8 @@ void *ipu_cpd_create_pkg_dir(struct ipu_bus_device *adev,
 	if (ret) {
 		dev_err(&isp->pdev->dev,
 			"Unable to parse module data section!\n");
-		dma_free_attrs(&isp->psys->dev, *pkg_dir_size, pkg_dir,
-			       *dma_addr,
-			       0
-		    );
+		dma_free_coherent(&isp->psys->dev, *pkg_dir_size, pkg_dir,
+			       *dma_addr);
 		return NULL;
 	}
 
@@ -258,7 +256,7 @@ void ipu_cpd_free_pkg_dir(struct ipu_bus_device *adev,
 			  u64 *pkg_dir,
 			  dma_addr_t dma_addr, unsigned int pkg_dir_size)
 {
-	dma_free_attrs(&adev->dev, pkg_dir_size, pkg_dir, dma_addr, 0);
+	dma_free_coherent(&adev->dev, pkg_dir_size, pkg_dir, dma_addr);
 }
 EXPORT_SYMBOL_GPL(ipu_cpd_free_pkg_dir);
 
diff --git a/drivers/media/pci/intel/ipu-isys.c b/drivers/media/pci/intel/ipu-isys.c
index e69b72a67c4d..41b95738419b 100644
--- a/drivers/media/pci/intel/ipu-isys.c
+++ b/drivers/media/pci/intel/ipu-isys.c
@@ -674,16 +674,14 @@ static void isys_remove(struct ipu_bus_device *adev)
 		debugfs_remove_recursive(isys->debugfsdir);
 
 	list_for_each_entry_safe(fwmsg, safe, &isys->framebuflist, head) {
-		dma_free_attrs(&adev->dev, sizeof(struct isys_fw_msgs),
-			       fwmsg, fwmsg->dma_addr,
-			       0
+		dma_free_coherent(&adev->dev, sizeof(struct isys_fw_msgs),
+			       fwmsg, fwmsg->dma_addr
 		    );
 	}
 
 	list_for_each_entry_safe(fwmsg, safe, &isys->framebuflist_fw, head) {
-		dma_free_attrs(&adev->dev, sizeof(struct isys_fw_msgs),
-			       fwmsg, fwmsg->dma_addr,
-			       0
+		dma_free_coherent(&adev->dev, sizeof(struct isys_fw_msgs),
+			       fwmsg, fwmsg->dma_addr
 		    );
 	}
 
@@ -786,10 +784,9 @@ static int alloc_fw_msg_buffers(struct ipu_isys *isys, int amount)
 					struct isys_fw_msgs, head);
 		list_del(&addr->head);
 		spin_unlock_irqrestore(&isys->listlock, flags);
-		dma_free_attrs(&isys->adev->dev,
+		dma_free_coherent(&isys->adev->dev,
 			       sizeof(struct isys_fw_msgs),
-			       addr, addr->dma_addr,
-			       0
+			       addr, addr->dma_addr
 		    );
 		spin_lock_irqsave(&isys->listlock, flags);
 	}
diff --git a/drivers/media/pci/intel/ipu-psys.c b/drivers/media/pci/intel/ipu-psys.c
index 06305d72b7e8..31d43aa3d3ba 100644
--- a/drivers/media/pci/intel/ipu-psys.c
+++ b/drivers/media/pci/intel/ipu-psys.c
@@ -1428,8 +1428,8 @@ static int ipu_psys_probe(struct ipu_bus_device *adev)
 	ipu_fw_com_release(psys->fwcom, 1);
 out_free_pgs:
 	list_for_each_entry_safe(kpg, kpg0, &psys->pgs, list) {
-		dma_free_attrs(&adev->dev, kpg->size, kpg->pg,
-			       kpg->pg_dma_addr, 0);
+		dma_free_coherent(&adev->dev, kpg->size, kpg->pg,
+			       kpg->pg_dma_addr);
 		kfree(kpg);
 	}
 
@@ -1480,8 +1480,8 @@ static void ipu_psys_remove(struct ipu_bus_device *adev)
 	mutex_lock(&ipu_psys_mutex);
 
 	list_for_each_entry_safe(kpg, kpg0, &psys->pgs, list) {
-		dma_free_attrs(&adev->dev, kpg->size, kpg->pg,
-			       kpg->pg_dma_addr, 0);
+		dma_free_coherent(&adev->dev, kpg->size, kpg->pg,
+			       kpg->pg_dma_addr);
 		kfree(kpg);
 	}
 
-- 
2.51.0

